src/tests/java/htsjdk/variant/variantcontext/GenotypeLikelihoodsUnitTest.java

```diff
@@ -182,22 +187,122 @@ public void testCalculatePLindex(){
-        allelePairTest(8, 2, 3);
-        allelePairTest(9, 3, 3);
+    @DataProvider
+    public Object[][] testGetAllelePairData() {
+        return new Object[][]{
+                {0, 0, 0},
+                {1, 0, 1},
+                {2, 1, 1},
+                {3, 0, 2},
+                {4, 1, 2},
+                {5, 2, 2},
+                {6, 0, 3},
+                {7, 1, 3},
+                {8, 2, 3},
+                {9, 3, 3}
+        };
     }
-        
-    private void allelePairTest(int PLindex, int allele1, int allele2) {
+
+    @Test(dataProvider = "testGetAllelePairData")
+    public void testGetAllelePair(final int PLindex, final int allele1, final int allele2){
         Assert.assertEquals(GenotypeLikelihoods.getAllelePair(PLindex).alleleIndex1, allele1, "allele index " + allele1 + " from PL index " + PLindex + " was not calculated correctly");
         Assert.assertEquals(GenotypeLikelihoods.getAllelePair(PLindex).alleleIndex2, allele2, "allele index " + allele2 + " from PL index " + PLindex + " was not calculated correctly");
     }
+
+    @DataProvider
+    public Object[][] testCalculatePLcacheData() {
+        return new Object[][]{
+                {
+                        2, 1,
+                        Arrays.asList(Arrays.asList(0),
+                                Arrays.asList(1)
+                        )
+                },
+                {
+                        2, 2,
+                        Arrays.asList(Arrays.asList(0, 0),
+                                Arrays.asList(0, 1),
+                                Arrays.asList(1, 1)
+                        )
+                },
+                {
+                        3, 2,
+                        Arrays.asList(Arrays.asList(0, 0),
+                                Arrays.asList(0, 1),
+                                Arrays.asList(1, 1),
+                                Arrays.asList(0, 2),
+                                Arrays.asList(1, 2),
+                                Arrays.asList(2, 2)
+                        )
+                },
+                {
+                        3, 3,
+                        Arrays.asList(Arrays.asList(0, 0, 0),
+                                Arrays.asList(0, 0, 1),
+                                Arrays.asList(0, 1, 1),
+                                Arrays.asList(1, 1, 1),
+                                Arrays.asList(0, 0, 2),
+                                Arrays.asList(0, 1, 2),
+                                Arrays.asList(1, 1, 2),
+                                Arrays.asList(0, 2, 2),
+                                Arrays.asList(1, 2, 2),
+                                Arrays.asList(2, 2, 2)
+                        )
+                }
+        };
+    }
+
+    @Test(dataProvider = "testCalculatePLcacheData")
+    public void testCalculatePLcache(final int numAlleles, final int ploidy,  final List<List<Integer>> expected){
+        GenotypeLikelihoods.plIndexToAlleleIndices.clear();
```

Comment:
Test should be using `initializePLIndexToAlleleIndices`, not replecating the code.